examples/vmwithcontracts/storage/recorder.go

```diff
@@ -0,0 +1,70 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package storage
+
+import (
+	"context"
+	"errors"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/utils/set"
+
+	"github.com/ava-labs/hypersdk/state"
+)
+
+type Recorder struct {
```

Comment:
(not for this PR) this would be a good utility function for the HyperSDK to provide

Reply:
What do you mean by "utility function"? You mean the whole struct?

Reply:
https://github.com/ava-labs/hypersdk/issues/1552